lib/internal/Magento/Framework/Module/DependencyChecker.php

```diff
@@ -89,17 +77,19 @@ public function checkDependenciesWhenEnableModules($toBeEnabledModules, $current
      * @param string[] $moduleNames list of modules to be enabled/disabled
      * @param string[] $enabledModules list of enabled modules assuming enable/disable succeeds
      * @return array
+     * @throws \Magento\Framework\Exception\LocalizedException
      */
     private function checkDependencyGraph($isEnable, $moduleNames, $enabledModules)
     {
-        $this->graph = $this->createGraph();
+        $fullModuleList = $this->loader->load();
+        $graph = $this->createGraph($fullModuleList);
         $dependenciesMissingAll = [];
         $graphMode = $isEnable ? Graph::DIRECTIONAL : Graph::INVERSE;
         foreach ($moduleNames as $moduleName) {
             $dependenciesMissing = [];
-            $paths = $this->graph->findPathsToReachableNodes($moduleName, $graphMode);
+            $paths = $graph->findPathsToReachableNodes($moduleName, $graphMode);
             $modules = array_merge(
```

Comment:
Would be great to remove array merge in loop, this will also speed up little bit

Reply:
It's not possible here. The result of modules merges used in algorithm

Reply:
Ah, yes, you’re right!